src/lib/comms/sol-lwm2m.c

```diff
@@ -4759,6 +5451,14 @@ sol_lwm2m_client_start(struct sol_lwm2m_client *client)
 
     return 0;
 
+err_access_control:
+    if (client->supports_access_control) {
+        ctx = find_object_ctx_by_id(client, ACCESS_CONTROL_OBJECT_ID);
+        while ((instance = sol_vector_get(&ctx->instances, 0)) != NULL) {
+            obj_instance_clear(client, ctx, instance);
+            sol_vector_del_element(&ctx->instances, instance);
+        }
+    }
```

Comment:
What about a case like this one: https://github.com/solettaproject/soletta/pull/2264/files#diff-e4058dc14a343a8aceb7210b323358a2R3174 ?

If I successfully delete 2 instances and fail when deleting the 3rd, it would still call `sol_vector_clear` outside the loop, trying to delete a `sol_vector` with elements still in it (at least the 3rd instance, in this case), and cause a bigger failure, wouldn't it? I mean, if I call `sol_vector_del_element` right after `obj_instance_clear`, it seems to me that I'm assuring more consistency.